readers/src/main/scala/com/salesforce/op/test/PassengerFeaturesTest.scala

```diff
@@ -34,36 +34,70 @@ import com.salesforce.op.features.types._
+    .extract(ageFnc)
+    .aggregate(TestMonoidAggregator)
     .asPredictor
 
-  val gender = FeatureBuilder.MultiPickList[Passenger].extract(p => Set(p.getGender).toMultiPickList).asPredictor
-  val genderPL = FeatureBuilder.PickList[Passenger].extract(p => p.getGender.toPickList).asPredictor
+  val gender = FeatureBuilder.MultiPickList[Passenger].extract(genderFnc).asPredictor
+  val genderPL = FeatureBuilder.PickList[Passenger].extract(genderPLFnc).asPredictor
 
   val height = FeatureBuilder.RealNN[Passenger]
-    .extract(p => Option(p.getHeight).map(_.toDouble).toRealNN(0.0))
+    .extract(heightFnc)
     .window(Duration.millis(300))
     .asPredictor
 
-  val heightNoWindow = FeatureBuilder.Real[Passenger].extract(_.getHeight.toReal).asPredictor
-  val weight = FeatureBuilder.Real[Passenger].extract(_.getWeight.toReal).asPredictor
-  val description = FeatureBuilder.Text[Passenger].extract(_.getDescription.toText).asPredictor
-  val boarded = FeatureBuilder.DateList[Passenger].extract(p => Seq(p.getBoarded.toLong).toDateList).asPredictor
-  val stringMap = FeatureBuilder.TextMap[Passenger].extract(p => p.getStringMap.toTextMap).asPredictor
-  val numericMap = FeatureBuilder.RealMap[Passenger].extract(p => p.getNumericMap.toRealMap).asPredictor
-  val booleanMap = FeatureBuilder.BinaryMap[Passenger].extract(p => p.getBooleanMap.toBinaryMap).asPredictor
-  val survived = FeatureBuilder.Binary[Passenger].extract(p => Option(p.getSurvived).map(_ == 1).toBinary).asResponse
-  val boardedTime = FeatureBuilder.Date[Passenger].extract(_.getBoarded.toLong.toDate).asPredictor
-  val boardedTimeAsDateTime = FeatureBuilder.DateTime[Passenger].extract(_.getBoarded.toLong.toDateTime).asPredictor
+  val heightNoWindow = FeatureBuilder.Real[Passenger].extract(heightToReal).asPredictor
+  val weight = FeatureBuilder.Real[Passenger].extract(weightToReal).asPredictor
+  val description = FeatureBuilder.Text[Passenger].extract(descrToText).asPredictor
+  val boarded = FeatureBuilder.DateList[Passenger].extract(boardedToDL).asPredictor
+  val stringMap = FeatureBuilder.TextMap[Passenger].extract(stringMapFnc).asPredictor
+  val numericMap = FeatureBuilder.RealMap[Passenger].extract(numericMapFnc).asPredictor
+  val booleanMap = FeatureBuilder.BinaryMap[Passenger].extract(booleanMapFnc).asPredictor
+  val survived = FeatureBuilder.Binary[Passenger].extract(survivedFnc).asResponse
+  val boardedTime = FeatureBuilder.Date[Passenger].extract(boardedTimeFnc).asPredictor
+  val boardedTimeAsDateTime = FeatureBuilder.DateTime[Passenger].extract(boardedDTFnc).asPredictor
 
   val rawFeatures: Array[OPFeature] = Array(
     survived, age, gender, height, weight, description, boarded, stringMap, numericMap, booleanMap
   )
 
 }
+
+object TestMonoidAggregator
+  extends CustomMonoidAggregator[Real](None, (l, r) => (l -> r).map(breeze.linalg.max(_, _))) with Serializable
+
+object PassengerFeaturesTestLambdas {
```

Comment:
do you need the functions defined this way or is it a style thing?

Reply:
that the new thing